CHANGELOG.md

```diff
@@ -1,8 +1,487 @@
+example, you can now run `ipfs swarm connect /dnsaddr/bootstrap.libp2p.io` to
+connect to one of the bootstrap peers at random. NOTE: This is could connect you
+to an _arbitrary_ peer as DNS is not secure (by default). Please do not rely on
+this except for testing or unless you know what you're doing.
+
+Finally, `ipfs swarm connect` now returns _all_ errors on failure. This should
+make it much easier to debug connectivity issues. For example, one might see an
+error like:
+
+```
+Error: connect QmYou failure: dial attempt failed: 6 errors occurred:
+	* <peer.ID Qm*Me> --> <peer.ID Qm*You> (/ip4/127.0.0.1/tcp/4001) dial attempt failed: dial tcp4 127.0.0.1:4001: connect: connection refused
+	* <peer.ID Qm*Me> --> <peer.ID Qm*You> (/ip6/::1/tcp/4001) dial attempt failed: dial tcp6 [::1]:4001: connect: connection refused
+	* <peer.ID Qm*Me> --> <peer.ID Qm*You> (/ip6/2604::1/tcp/4001) dial attempt failed: dial tcp6 [2604::1]:4001: connect: network is unreachable
+	* <peer.ID Qm*Me> --> <peer.ID Qm*You> (/ip6/2602::1/tcp/4001) dial attempt failed: dial tcp6 [2602::1]:4001: connect: network is unreachable
+	* <peer.ID Qm*Me> --> <peer.ID Qm*You> (/ip4/150.0.1.2/tcp/4001) dial attempt failed: dial tcp4 0.0.0.0:4001->150.0.1.2:4001: i/o timeout
+	* <peer.ID Qm*Me> --> <peer.ID Qm*You> (/ip4/200.0.1.2/tcp/4001) dial attempt failed: dial tcp4 0.0.0.0:4001->200.0.1.2:4001: i/o timeout
+```
+
+#### Changed: `ipfs bitswap stat`
+
+`ipfs bitswap stat` no longer lists bitswap partners unless the `-v` flag is
+passed. That is, it will now return:
+
+```
+> ipfs bitswap stat
+bitswap status
+	provides buffer: 0 / 256
+	blocks received: 0
+	blocks sent: 79
+	data received: 0
+	data sent: 672706
+	dup blocks received: 0
+	dup data received: 0 B
+	wantlist [0 keys]
+	partners [197]
+```
+
+Instead of:
+
+```
+> ipfs bitswap stat -v
+bitswap status
+	provides buffer: 0 / 256
+	blocks received: 0
+	blocks sent: 79
+	data received: 0
+	data sent: 672706
+	dup blocks received: 0
+	dup data received: 0 B
+	wantlist [0 keys]
+	partners [203]
+		QmNQTTTRCDpCYCiiu6TYWCqEa7ShAUo9jrZJvWngfSu1mL
+		QmNWaxbqERvdcgoWpqAhDMrbK2gKi3SMGk3LUEvfcqZcf4
+		QmNgSVpgZVEd41pBX6DyCaHRof8UmUJLqQ3XH2qNL9xLvN
+        ... omitting 200 lines ...
+```
+
+#### Changed: `ipfs repo stat --human`
+
+The `--human` flag in the `ipfs repo stat` command now intelligently picks a
+size unit instead of always using MiB.
+
+#### Changed: `ipfs resolve` (`ipfs dns`, `ipfs name resolve`)
+
+All of the resolve commands now:
+
+1. Resolve _recursively_ (up to 32 steps) by default to better match user
+   expectations (these commands used to be non-recursive by default). To turn
+   recursion off, pass `-r false`.
+2. When resolving non-recursively, these commands no longer fail when partially
+   resolving a name. Instead, they simply return the intermediate result.
+
+#### Changed: `ipfs files flush`
+
+The `ipfs files flush` command now returns the CID of the flushed file.
```

Comment:
cc @achingbrain

Reply:
I _believe_ we did this to improve the CoreAPI. IIRC, my end goal was to avoid exposing any CIDs to the user until they explicitly call `Flush` so we can make the actual IPFS import part lazy.